packages/devtools_app/lib/src/shared/diagnostics/variable_factory.dart

```diff
@@ -478,6 +478,57 @@ List<DartObjectNode> createVariablesForElements(
   return variables;
 }
 
+List<DartObjectNode> createVariablesForRecords(
+  Instance instance,
+  IsolateRef? isolateRef,
+) {
+  final positionalFields = <BoundField>[];
+  final namedFields = <BoundField>[];
+  for (final field in instance.fields ?? []) {
+    if (_isPositionalField(field)) {
+      positionalFields.add(field);
+    } else {
+      namedFields.add(field);
+    }
+  }
+  // Sort positional fields in ascending order:
+  _sortPositionalFields(positionalFields);
+  final variables = <DartObjectNode>[];
+  // Always show positional fields before named fields:
+  for (final field in positionalFields) {
+    variables.add(
+      DartObjectNode.fromValue(
+        // Positional fields are designated by their getter syntax, eg $0, $1,
+        // $2, etc:
+        name: '\$${field.name}',
+        value: field.value,
+        isolateRef: isolateRef,
+      ),
+    );
+  }
+  for (final field in namedFields) {
+    variables.add(
+      DartObjectNode.fromValue(
+        name: field.name,
+        value: field.value,
+        isolateRef: isolateRef,
+      ),
+    );
+  }
```

Comment:
this can all be inlined in the list:
```
final variables = [
  // Always show positional fields before named fields:
  for (final field in positionalFields)
    DartObjectNode.fromValue(
        // Positional fields are designated by their getter syntax, eg $0, $1,
        // $2, etc:
        name: '\$${field.name}',
        value: field.value,
        isolateRef: isolateRef,
      ),
  ...
];
```

Reply:
Good point, done!